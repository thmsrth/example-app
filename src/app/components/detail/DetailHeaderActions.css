.DetailHeaderActions {
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
}

.DetailHeaderActions__container {
  display: inline-flex;
}

.DetailHeaderActions__button {
  margin-left: var(--section-margin);
  color: var(--color-white);
  background-color: var(--color-action);
}

.DetailHeaderActions__button--disabled {
  color: var(--button-disabled-color);
  background-color: var(--button-disabled-background-color);
}

.DetailHeaderActions__button__delete {
  margin-left: var(--section-margin);
  background-color: var(--action-delete-background-color);
  color: var(--action-delete-color);
}

.DetailHeaderActions__button__delete--disabled {
  color: var(--button-disabled-color);
  background-color: var(--button-disabled-background-color);
}
