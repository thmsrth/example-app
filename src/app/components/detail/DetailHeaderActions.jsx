import React from 'react';

import PropTypes from 'prop-types';

import Button from '../../../global/components/Button';

import { canBeDeleted } from '../../utils/BusinessLogic';

import I18n from '../../utils/I18n';
import SecurityUtil from '../../../global/security/utils/SecurityUtil';

import dataItem from '../types/dataItem';

import './DetailHeaderActions.css';

const DetailHeaderActions = ({
  handleEditData, handleDeleteData, dataDeleting, item,
}) => (
  <div className="DetailHeaderActions">
    <div className="DetailHeaderActions__container">
      <Button
        className="DetailHeaderActions__button"
        text={I18n.t('en.actions.disabled')}
        iconName="beer"
        onClick={() => {}}
        disabled
      />

      <Button
        className="DetailHeaderActions__button"
        text={I18n.t('en.actions.edit')}
        iconName="edit"
        onClick={handleEditData}
        disabled={!SecurityUtil.isGranted('EXAMPLE_EDIT')}
      />

      {canBeDeleted(item) && (
        <Button
          className="DetailHeaderActions__button DetailHeaderActions__button__delete"
          text={I18n.t('en.actions.delete')}
          iconName="trash"
          onClick={handleDeleteData}
          loading={dataDeleting !== ''}
          disabled={!SecurityUtil.isGranted('EXAMPLE_DELETE')}
        />
      )}
    </div>
  </div>
);

export default DetailHeaderActions;

DetailHeaderActions.propTypes = {
  handleEditData: PropTypes.func,
  handleDeleteData: PropTypes.func,
  dataDeleting: PropTypes.string,
  item: dataItem,
};

DetailHeaderActions.defaultProps = {
  handleEditData: () => {},
  handleDeleteData: () => {},
  dataDeleting: '',
  item: null,
};
