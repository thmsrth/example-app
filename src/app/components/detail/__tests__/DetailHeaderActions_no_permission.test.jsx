import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import DetailHeaderActions from '../DetailHeaderActions';

global.permissions = {};

const props = {
  item: {
    status: 'UNHEALTHY',
  },
};

describe('DetailHeaderActions', () => {
  it('should render the buttons disabled', () => {
    const tree = shallow(<DetailHeaderActions {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
