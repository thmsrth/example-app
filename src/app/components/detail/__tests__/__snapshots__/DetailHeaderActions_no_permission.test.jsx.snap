// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailHeaderActions should render the buttons disabled 1`] = `
<div
  className="DetailHeaderActions"
>
  <div
    className="DetailHeaderActions__container"
  >
    <ButtonWrapper
      className="DetailHeaderActions__button"
      disabled={true}
      iconName="beer"
      onClick={[Function]}
      params={Object {}}
      text="Disabled action"
    />
    <ButtonWrapper
      className="DetailHeaderActions__button"
      disabled={true}
      iconName="edit"
      onClick={[Function]}
      params={Object {}}
      text="Edit"
    />
    <ButtonWrapper
      className="DetailHeaderActions__button DetailHeaderActions__button__delete"
      disabled={true}
      iconName="trash"
      loading={false}
      onClick={[Function]}
      params={Object {}}
      text="Delete"
    />
  </div>
</div>
`;
